engine/src/main/java/org/pentaho/di/pan/Pan.java

```diff
@@ -239,48 +244,86 @@ public static void main( String[] a ) throws Exception {
-              .localJarFile( optionJarFilename.toString() )
-              .localInitialDir( initialDir.toString() )
-              .listRepos( optionListrep.toString() )
-              .safeMode( optionSafemode.toString() )
-              .metrics( optionMetrics.toString() )
-              .listFileParams( optionListParam.toString() )
-              .logLevel( "" )
-              .maxLogLines( "" )
-              .maxLogTimeout( "" )
-              .logFile( "" )
-              .oldLogFile( "" )
-              .version( "" )
-              .resultSetStepName( optionResultSetStepName.toString() )
-              .resultSetCopyNumber( optionResultSetCopyNumber.toString() )
-              .base64Zip( optionBase64Zip.toString() )
-              .namedParams( optionParams )
-              .build();
-
-      Result result = getCommandExecutor().execute( transParams, args.toArray( new String[ args.size() ] ) );
-
-      exitJVM( result.getExitStatus() );
+        .blockRepoConns( optionNorep.toString() )
+        .repoName( optionRepname.toString() )
+        .repoUsername( optionUsername.toString() )
+        .trustRepoUser( optionTrustUser.toString() )
+        .repoPassword( optionPassword.toString() )
+        .inputDir( optionDirname.toString() )
+        .inputFile( optionTransname.toString() )
+        .listRepoFiles( optionListtrans.toString() )
+        .listRepoDirs( optionListdir.toString() )
+        .exportRepo( optionExprep.toString() )
+        .localFile( optionFilename.toString() )
+        .localJarFile( optionJarFilename.toString() )
+        .localInitialDir( initialDir.toString() )
+        .listRepos( optionListrep.toString() )
+        .safeMode( optionSafemode.toString() )
+        .metrics( optionMetrics.toString() )
+        .listFileParams( optionListParam.toString() )
+        .logLevel( "" )
+        .maxLogLines( "" )
+        .maxLogTimeout( "" )
+        .logFile( "" )
+        .oldLogFile( "" )
+        .version( "" )
+        .resultSetStepName( optionResultSetStepName.toString() )
+        .resultSetCopyNumber( optionResultSetCopyNumber.toString() )
+        .base64Zip( optionBase64Zip.toString() )
+        .namedParams( optionParams )
+        .build();
+
+      Result rslt = getCommandExecutor().execute( transParams, args.toArray( new String[ args.size() ] ) );
+
+      exitJVM( rslt.getExitStatus() );
 
     } catch ( Throwable t ) {
       t.printStackTrace();
       exitJVM( CommandExecutorCodes.Pan.UNEXPECTED_ERROR.getCode() );
     }
   }
 
+  private static CommandExecutorResult validateAndSetPluginParam( LogChannelInterface log, String path ) {
+//    String returnMsg = "";
+    CommandExecutorResult result = null;
+      try {
+        Collection<CommandLineOptionProvider> commandlineProviders = PluginServiceLoader.loadServices( CommandLineOptionProvider.class );
+        CommandLineOptionProvider provider = commandlineProviders.stream().findFirst().orElse( null );
```

Comment:
why are we only using one? Shouldn't we use all the plugin-provided CommandLineOptions?